lib/experimental/class-wp-navigation-fallbacks-gutenberg.php

```diff
@@ -0,0 +1,223 @@
+				'post_status'  => 'publish',
+				'post_type'    => 'wp_navigation',
+			),
+			true // So that we can check whether the result is an error.
+		);
+
+		return $wp_insert_post_result;
+	}
+
+	/**
+	 * Determine the most appropriate classic navigation menu to use as a fallback.
+	 *
+	 * @return object WP_Term The classic navigation.
+	 */
+	public static function get_classic_menu_fallback() {
+		$classic_nav_menus = wp_get_nav_menus();
+
+		if ( ! $classic_nav_menus || is_wp_error( $classic_nav_menus ) ) {
+			return null;
+		}
+
+		// 1.
+		// Attempt to use the menu assigned to location `primary`.
+		$nav_menu = static::get_nav_menu_at_primary_location();
+
+		if ( $nav_menu ) {
+			return $nav_menu;
+		}
+
+		// 2.
+		// Use the menu with `primary` as its slug.
+		$nav_menu = static::get_nav_menu_with_primary_slug( $classic_nav_menus );
+
+		if ( $nav_menu ) {
+			return $nav_menu;
+		}
+
+		// 3.
+		// Otherwise use the most recently created classic menu.
+		return static::get_most_recently_created_nav_menu( $classic_nav_menus );
+	}
+
+
+	/**
+	 * Sorts the classic menus and returns the most recently created one.
+	 *
+	 * @param WP_Term[] $classic_nav_menus Array of classic nav menu term objects.
+	 * @return WP_Term The most recently created classic nav menu.
+	 */
+	private static function get_most_recently_created_nav_menu( $classic_nav_menus ) {
+		usort(
+			$classic_nav_menus,
+			function( $a, $b ) {
+				return $b->term_id - $a->term_id;
+			}
+		);
+
+		return $classic_nav_menus[0];
+	}
+
+	/**
+	 * Returns the classic menu with the slug `primary` if it exists.
+	 *
+	 * @param WP_Term[] $classic_nav_menus Array of classic nav menu term objects.
+	 * @return WP_Term|null The classic nav menu with the slug `primary` or null.
+	 */
+	private static function get_nav_menu_with_primary_slug( $classic_nav_menus ) {
+		foreach ( $classic_nav_menus as $classic_nav_menu ) {
+			if ( 'primary' === $classic_nav_menu->slug ) {
+				return $classic_nav_menu;
+			}
+		}
+
+		return null;
+	}
+
+
+	/**
+	 * Gets the classic menu assigned to the `primary` navigation menu location
+	 * if it exists.
+	 *
+	 * @return WP_Term|null The classic nav menu assigned to the `primary` location or null.
+	 */
+	private static function get_nav_menu_at_primary_location() {
+		$locations = get_nav_menu_locations();
+
+		if ( isset( $locations['primary'] ) ) {
+			$primary_menu = wp_get_nav_menu_object( $locations['primary'] );
+
+			if ( $primary_menu ) {
+				return $primary_menu;
+			}
+		}
+
+		return null;
+	}
+
+	/**
+	 * Creates a default Navigation Menu fallback.
+	 *
+	 * @return int|WP_Error The post ID of the default fallback menu or a WP_Error object.
+	 */
+	private static function create_default_fallback() {
+		$registry = WP_Block_Type_Registry::get_instance();
+
+		// If `core/page-list` is not registered then use empty blocks.
+		$default_blocks = $registry->is_registered( 'core/page-list' ) ? '<!-- wp:page-list /-->' : '';
```

Comment:
Should we make this a class constant so its easier to update?

Reply:
✅